src/univariate/discrete/dirichletcategorical.jl

```diff
@@ -0,0 +1,120 @@
+"""
+    DirichletCategorical(α,n)
+
+A *Dirichlet-categorical distribution* is the compound distribution of the [`Categorical`](@ref) distribution where the probability of each category is distributed according to the [`Dirichlet`](@ref). It has two parameters: the concentration parameters `α` and the optional observation counts `n`.
+
+```math
+p(x = i \\vert n, \\alpha) = \\int\\limits_p p(x \\vert p) p(p \\vert n, \\alpha) dp = \\frac{n_i + \\alpha_1}{\\sum\\limits_k n_k + α_k}
+```
+
+```julia
+DirichletCategorical(α, n)      # DirichcletCategorical distribution with concentration parameters α and observation counts n.
+
+params(d)       # Get the parameters, i.e. (α, n)
+update!(d, o)   # Update the counts d.n with observation(s) o.
+```
+
+External links:
+
+* [Categorical distribution conjugate prior on Wikipedia](https://en.wikipedia.org/wiki/Categorical_distribution#Bayesian_inference_using_conjugate_prior)
+"""
+struct DirichletCategorical{T <: Real, U <: Integer} <: DiscreteUnivariateDistribution
+    α::Vector{T}
+    α0::T
+    n::Vector{U}
+
+    function DirichletCategorical(α::Vector{T}, n::Vector{U}) where {T, U}
```

Comment:
Could we use akeyword argument  `check_args=true` and check arguments only if this is true? This allows users to skip the check if they know their construction is valid